golang/pkg/events/events.go

```diff
@@ -1,24 +1,55 @@
 package events
 
 import (
+	"time"
+
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func ToEvent(event proto.Message, source string, id string) ([]byte, error) {
+type cloudEventOptions struct {
+	id     string
+	source string
+	time   *time.Time
+}
+
+type Option = func(fields *cloudEventOptions)
+
+func WithTime(time *time.Time) Option {
+	return func(fields *cloudEventOptions) {
+		fields.time = time
+	}
+}
+
+func ToEvent(event proto.Message, id string, source string, opts ...Option) ([]byte, error) {
+	fields := &cloudEventOptions{id: id, source: source}
```

Comment:
I actually liked the option to have default `id` and `source`, specially the 1st one. We have the same in elixir.
By the way, what happens if the user doesn't provide the time. Where is the default one created?